### PR TITLE
Improved logging at activation of new VCL

### PR DESCRIPTION
**Pull request by robert7**
We had a suspicion, that sometimes the VCL was not updated correctly or maybe not at all replicas. Therefore, I added more detailed logging, which enables better examination/troubleshooting. No logs are added in default log level. New logs only show up, if logging is explicitly set to higher level.

Polling watch loop shows timestamps of watched file and 1x per hour also md5 hash current file contents (this also confirms the watch loop is still alive). At the template rendering time, again md5 sum of unparsed template is logged, together with template parameters. Note that the default fsnotify version of the watch doesn't work at all in our setup (VCL text set by configmap and mapped as a volume into the container with varnish).

Also fixed some typos in flag description and added short note about activation of detailed logging in README.

### CODE REVIEW

File: pkg/controller/watch.go [hunk 1]
```diff
@@ -57,6 +56,17 @@ func (v *VarnishController) watchConfigUpdates(ctx context.Context, c *exec.Cmd,
...
 
+func (v *VarnishController) setTemplate(tmplContents []byte) error {
+	parsedTemplate, err := template.New("vcl").Parse(string(tmplContents))
+	if err == nil {
+		v.vclTemplate = parsedTemplate
+		hash := md5.Sum(tmplContents)
+		hashStr := hex.EncodeToString(hash[:])
+		v.vclTemplateHash = hashStr
+	}
+	return err
```
Comment (martin-helmich) [inline pkg/controller/watch.go:67]: This is a matter of preference, but I think the usual `if err != nil`-approach might be more intuitive, here.

```suggestion
	parsedTemplate, err := template.New("vcl").Parse(string(tmplContents))
	if err != nil {
		return err
	}

	v.vclTemplate = parsedTemplate
	hash := md5.Sum(tmplContents)
	hashStr := hex.EncodeToString(hash[:])
	v.vclTemplateHash = hashStr

	return nil
```

File: pkg/watcher/template_watch_poll.go [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
 	"io/ioutil"
 	"os"
 	"time"
 )
 
+const (
+	// how often to check for template file changes
+	POLL_INTERVAL_SECONDS = 15
+	// how often print template info for troubleshooting
+	TIMESTAMP_DISPLAY_INTERVAL_SECONDS = 3600
```
Comment (martin-helmich) [inline pkg/watcher/template_watch_poll.go:16]: We can use the `time.Duration` type here, directly.

```suggestion
	// how often to check for template file changes
	POLL_INTERVAL = 15 * time.Second
	// how often print template info for troubleshooting
	TIMESTAMP_DISPLAY_INTERVAL = 1 * time.Hour
```

**Review by martin-helmich — APPROVED** [review]
Sorry for the delay and thanks for the contribution. :+1:  This looks good to me; I've only left some stylistic suggestions.

File: pkg/controller/watch.go (hunk 1 above)
Comment (robert7) [reply pkg/controller/watch.go]: done

File: pkg/watcher/template_watch_poll.go (hunk 2 above)
Comment (robert7) [reply pkg/watcher/template_watch_poll.go]: I applied both requested changes, pls review again

**Comment by robert7** [comment]
Please recheck my changes...

**Review by martin-helmich — APPROVED** [review]
Apologies for the delay (FWIW, we've also disabled the "stale" workflow, since it's doing more harm than good). This looks good to me now. :+1:  Thanks for your contribution.

**Comment by robert7** [comment]
OK thanks

**Pull request merged**